### PR TITLE
Proxy cache procedure

### PR DESCRIPTION
**Pull request by stevsmit**

### CODE REVIEW

File: modules/proxy-namespace-procedure.adoc [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+* The default *Expiration* for cached images is set to 86400 seconds. This value can be adjusted. but is refreshed every pull. Un-refreshed images will disappear after the allotted time, but are still stored in Quay. The time in which an image is completely deleted, or garbage collected, depends on the *Time Machine* setting of your organization. The default time for garbage collection is 14 days unless otherwise specified.
+====
+
+. Optional: Click *Add Storage Quota* to configure quota management for your organization. For more information about quota management, see link:insert_link_here[Quota Management].
++
+[NOTE]
+====
+If your organization is configured with low quota and set to reject images that exceed the established quota, the following error is returned: `unable to pull image: Error parsing image configuration: Error fetching blob: invalid status code from registry 403 (Forbidden)`. Error `403` is inaccurate. Users should instead be told that their allotted quota has been exceeded upon rejection. This known issue will be fixed in a future update.
```
Comment (flavianmissi) [inline modules/proxy-namespace-procedure.adoc:30]: This is not the case for all pulls where the quota limit is exceeded. The reason is that if the quota limit is exceeded _during_ the pull podman will hide the error the API sends and show the generic 403. 
In contrast, when the quota limit is found to be exceeded on the first operation of the pull then the correct message is shown to the user (Quota limit exceeded).

Recall that a pull is compose of several different http calls to the Quay API, notably:
1. pull the image manifest
1. pull the image config and blobs
Image pulls always follow the above order.
In other words, when Quay returns the quota exceeded error on the manifest pull step, podman will show the correct "Quota limit exceeded" message. But when Quay returns the quota exceeded error on step 2, podman will show the generic 403 error message, despite of Quay's API returning the specific quota exceeded error.

You may want to keep the docs simple and just say that podman MAY show a 403 if the quota limit is reached _during_ a pull, hiding the original quota message.

File: modules/proxy-namespace-procedure.adoc [hunk 2]
```diff
@@ -0,0 +1,42 @@
+
+[[red-hat-quay-proxy-namespace-procedure]]
+= Using {productname} to proxy a namespace in a registry
+
+The following procedure describes how you can use {productname} to proxy a namespace in a registry. This procedure is set up to proxy quay.io, which allows users to use `podman` to pull any public image from any namespace on quay.io.
```
Comment (flavianmissi) [inline modules/proxy-namespace-procedure.adoc:5]: In both this sentence and on the title right above it we're saying that this document will show how to proxy a namespace in a registry.
The examples then show how to proxy an entire registry, and in the notes is where we mention that you can also proxy a single namespace.
This feels a little confusing to me, do you think we can make this more consistent?

File: modules/proxy-namespace-procedure.adoc [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+.Prerequisites
+
+*
```
Comment (flavianmissi) [inline modules/proxy-namespace-procedure.adoc:9]: Should we mention here that the [FEATURE_PROXY_CACHE](https://github.com/quay/quay/blob/master/config.py#L47) config needs to set to `true` in the `config.yaml` for this feature to become available?
Even our own QE was confused by that when I failed to mention this config in the testing steps I gave them 😅

File: modules/proxy-namespace-procedure.adoc [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+
+. In your Quay organization on the UI, for example, `cache-quayio`, click *Organization Settings* on the left hand pane.
+
+. In *Remote Registry* enter the name of the remote registry to be cached, for example, `quay.io`, and click *Save*.
++
+[NOTE]
+====
+* By adding a namespace to the *Remote Registry*, for example, `quay.io/<namespace>`, users in your organization will only be able to proxy from that namespace.
+* If you do not set a *Remote Registry Username* and *Remote Registry Password*, you cannot add one without removing the proxy cache and creating a new registry.
+* The default *Expiration* for cached images is set to 86400 seconds. This value can be adjusted. but is refreshed every pull. Un-refreshed images will disappear after the allotted time, but are still stored in Quay. The time in which an image is completely deleted, or garbage collected, depends on the *Time Machine* setting of your organization. The default time for garbage collection is 14 days unless otherwise specified.
```
Comment (flavianmissi) [inline modules/proxy-namespace-procedure.adoc:23]: Can you elaborate on the sentence below?
> This value can be adjusted. but is refreshed every pull.

The "but" here has me thinking that even though the user may set an expiration other than the default 86400, Quay will overwrite it with the default 86400 on every pull.

In case there's confusion on how this works (because I think this is confusing): "expiration" is the default _tag_ expiration Quay sets for all cached images. The tag expiration is renewed (reset, or refreshed) every time the tag is pulled. It's possible to override individual tag expiration (this is a Quay feature, independent of proxy), but since Quay renews this value based on the proxy org's "expiration" config, individual tag expiration overrides will only last until the tag's next pull, when Quay will reset it to whatever the proxy org's "expiration" config is set to.

Also: you totally nailed the interaction of the cache with the time machine config! 🚀

Comment (flavianmissi) [inline modules/proxy-namespace-procedure.adoc:23]: Oh and consider using the word `stale` rather than `un-refreshed`. In the future we'll have a field that will refer to staleness, so we might as well introduce the term now.

Comment (flavianmissi) [reply modules/proxy-namespace-procedure.adoc:23]: Or actually `expired` would be better suited, since we're talking about tag expiration. And it's a term we already use.

**Review by flavianmissi — APPROVED** [review]
LGTM!
